### PR TITLE
Introduce util function to concatenate queries

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Fixes: [1734](https://github.com/ballerina-platform/ballerina-standard-library/issues/1734) & [1886](https://github.com/ballerina-platform/ballerina-standard-library/issues/1886)

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

**Comment by daneshk** [comment]
@kalaiyarasiganeshalingam shall we add a test case for IN clause as well. where we have an array of values and construct a parameterized query by iterating through the array

File: changelog.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+- Introduce util function to concatenate queries
```
Comment (niveathika) [inline changelog.md:10]: Lets create an issue and link to it.

File: ballerina/tests/simple-params-query-test.bal [hunk 2]
```diff
@@ -780,6 +780,37 @@ function queryArrayBasicNullParams() returns error? {
...
+    record{}? returnData = check queryMockClient(simpleParamsDb, sqlQuery);
+    test:assertEquals(returnData["TOTAL"], 3, "Update command was not successful.");
+}
+
+@test:Config {
+    groups: ["query", "query-simple-params"]
+}
+function testInOperator3() returns error? {
+    int[] ids = [1,2,3];
+    ParameterizedQuery sqlQuery = `SELECT count(*) as total FROM DataTable WHERE row_id in (${ids[0]}, ${ids[1]}, ${ids[2]})`;
```
Comment (niveathika) [inline ballerina/tests/simple-params-query-test.bal:809]: Shall we improve this use case? Assuming the array length can vary. A user has to use a query to concat the final query. Let's change this test case for that.

File: ballerina/types.bal [hunk 3]
```diff
@@ -1408,3 +1408,45 @@ public type CustomResultIterator object {
...
+}
+
+function prepareParameterizedQuery(ParameterizedQuery[] queries) returns ParameterizedQuery {
+    ParameterizedQuery newParameterQuery = ``;
+    string queryInString = "";
+    string nullValue = "";
+    string[] strings = [];
+    Value[] values = [];
+    foreach ParameterizedQuery query in queries {
+        string[] stringValues = query.strings;
```
Comment (niveathika) [inline ballerina/types.bal:1429]: Shall we move this to util.bal? Technically these are not types

File: ballerina/tests/simple-params-query-test.bal [hunk 4]
```diff
@@ -780,6 +780,124 @@ function queryArrayBasicNullParams() returns error? {
     }
 }
 
+@test:Config {
+    groups: ["query", "query-simple-params"]
+}
+function testInOperator1() returns error? {
+    int[] ids = [1,2,3];
+    ParameterizedQuery sqlQuery = queryConcat(`SELECT count(*) as total FROM DataTable`,
+                                              ` WHERE row_id in (${ids[0]}, ${ids[1]}, ${ids[2]})`);
```
Comment (niveathika) [inline ballerina/tests/simple-params-query-test.bal:789]:
```suggestion
    ParameterizedQuery sqlQuery = queryConcat(`SELECT count(*) as total FROM DataTable WHERE row_id in (`,
                                              `${ids[0]}, ${ids[1]}, ${ids[2]})`);
```

File: ballerina/tests/simple-params-query-test.bal (hunk 2 above)
Comment (niveathika) [reply ballerina/tests/simple-params-query-test.bal:808]: arrayFlattenQuery() solves this.

File: ballerina/utils.bal [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                foreach var inValue in inValues {
+                    stingArray += convertToInValue(inValue);
+                }
+                newQuery += 'string:substring(stingArray, 0, stingArray.length() - 1) + "],";
+            } else if (inValues is anydata) {
+                newQuery += convertToInValue(inValues);
+            }
+        } else {
+            newQuery += convertToInValue(value);
+        }
```
Comment (daneshk) [inline ballerina/utils.bal:56]: Can't we use the `queryConcat()` function to construct the parameterized query?

File: ballerina/utils.bal [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        } else {
+            values.push(insertionValue);
+        }
+    }
+}
+# Flattens the array.
+#
+# + values - An array of `Value`
+# + return - A `ParameterizedQuery`
+public function arrayFlattenQuery(Value[] values) returns ParameterizedQuery {
```
Comment (daneshk) [inline ballerina/utils.bal:88]:
```suggestion
public isolated function arrayFlattenQuery(Value[] values) returns ParameterizedQuery {
```

File: ballerina/utils.bal [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/lang.'string;
+
+# Concatenates all the queries.
+#
+# + queries - Set of `ParameterizedQuery`
+# + return - A `ParameterizedQuery`
+public function queryConcat(ParameterizedQuery... queries) returns ParameterizedQuery {
```
Comment (daneshk) [inline ballerina/utils.bal:23]:
```suggestion
public isolated function queryConcat(ParameterizedQuery... queries) returns ParameterizedQuery {
```

Comment (daneshk) [reply ballerina/utils.bal:23]: shall we make it isolated

File: ballerina/tests/simple-params-query-test.bal (hunk 4 above)
Comment (kalaiyarasiganeshalingam) [reply ballerina/tests/simple-params-query-test.bal:789]: Added new testcase for this. https://github.com/ballerina-platform/module-ballerina-sql/pull/367/files#diff-e2e5a38967208c0c5bd499529715b43f511030da68452e7e70dabcd43cd42a62R916

File: ballerina/utils.bal (hunk 5 above)
Comment (kalaiyarasiganeshalingam) [reply ballerina/utils.bal]: Updated

**Review by daneshk — APPROVED** [review]
LGTM

**Pull request merged**